packages/react/src/components/Tree/Tree.tsx

```diff
@@ -263,32 +266,42 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
           const { elementRef, ...restItemForRender } = itemForRender
           const isSubtree = hasSubtree(item)
           const isSubtreeOpen = isSubtree && this.isActiveItem(item['id'])
-
-          const renderedItem = TreeItem.create(item, {
-            defaultProps: {
-              className: Tree.slotClassNames.item,
-              open: isSubtreeOpen,
-              renderItemTitle,
-              key: item['id'],
-              ...restItemForRender,
-            },
-            overrideProps: this.handleTreeItemOverrides,
-          })
-
-          // Only need refs of the items that spawn subtrees, when they need to be focused
-          // by any of their children, using Arrow Left.
-          const finalRenderedItem = isSubtree ? (
+          const renderedItem = (
+            <Ref key={item['id']} innerRef={elementRef}>
+              {TreeItem.create(item, {
```

Comment:
This line does 3 things: 
- resolve user's `item` value into ElementType and props
- Mix in defaultProps and overrideProps
- Create an element from the final ElementType and mixed props

It needs to only do items 1 and 2, get final ElementType and props.  The creation of the JSX should be done at a later phase by the virtualized step where it can mix in additional props, like style.

There needs to be some prop or pattern available to the user to get the computed ElementType and props before the JSX is created.  It should also allow them to create the JSX themselves.